src/modules/form/hooks/useViewEditRecordDialogs.tsx

```diff
@@ -18,6 +18,7 @@ interface Args<T> {
   evictCache?: VoidFunction;
   localConstants?: LocalConstants;
   maxWidth?: Breakpoint | false;
+  projectId?: string; // Project context for fetching form definition
```

Comment:
I had to pass this through a looot of components to get where it has to go for `useFormDefinition`. In other places we pass the project id through localConstants or inputVariables but I wanted to add a way to make it explicit.

I think we should move the form defintion fetching up, but that was a big refactor so i didnt do it here